apache-demo/docs/adopt.md

```diff
@@ -0,0 +1,177 @@
+    "os_auth_project_id": "701600a059af40e1865a3c494288712a"
+    ```
+    8. Click Save.
+4. Upload the Apache Demo Resource Package. The resource package contains the apache-demo assembly descriptor, the resource descriptor for the Apache2 server, a heat template, and configuration files. For an example resource package, see IBM/tnco-package-examples/Apache-demo.  
+You upload the demo resource package to IBM Telco Network Cloud - Orchestration by using the lmctl command-line tool. For more information about the lmctl command-line tool, see https://github.com/IBM/lmctl).  
+To upload the demo resource package, follow these steps:  
+    1. Unpack the package-examples on the client computer where lmctl is installed and configured.
+    2. Browse to the tnco-package-examples/apache-demo directory.
+    3. Push the assembly to IBM Telco Network Cloud - Orchestration with the following command:
+    ```
+    lmctl project push env
+    ```
+    Where env is the name of your IBM Telco Network Cloud - Orchestration environment that you set up in your lmctl configuration.  
+  
+5. Adopt the stack by way of the REST API. For this procedure, use the curl CLI commands to access the IBM Telco Network Cloud - Orchestration API.
+To access the REST API, retrieve an Access Token for Authorization.
+
+    **Note:** The first three steps must be done on the OCP system, or a client where you have access to the OC command to get the CLIENT_SECRET. The subsequent steps can be continued from there or from another client.
+    
+    1. Set the following initial environment variables: the name of the namespace, the name of the secret that holds the IBM Telco Network Cloud - Orchestration credentials, and the CLIENT_ID as Admin:
+    ```
+    NAMESPACE=lifecycle-manager
+    TNCO_CREDENTIALS_SECRET=alm-credentials-default
+    CLIENT_ID=Admin
+    ```
+    2. Get the $CLIENT_SECRET is the key for clientId: Admin and decode it using base64.
+    ```
+    CLIENT_SECRET=$(oc get secret $TNCO_CREDENTIALS_SECRET -o jsonpath='{.data.adminClientSecret}' -n $NAMESPACE | base64 -d)
+    ```
+    3. Fetch the API PROXY_ADDRESS (Ishtar route):
+    ```
+    PROXY_ADDRESS=$(oc get routes alm-ishtar -o jsonpath='{.spec.host}' -n $NAMESPACE)
+    ```
+    4. Set an encoded header for basic authorization:
+    ```
+    BASIC_AUTH_ENCODED=$(echo -n "$CLIENT_ID:$CLIENT_SECRET" | base64)
+    ```
+    5. Generate the ACCESS_TOKEN that is required for OAuth authentication:
+    ```
+    ACCESS_TOKEN=$(curl -sk -X POST "https://$PROXY_ADDRESS/oauth/token" -H "Authorization: Basic $BASIC_AUTH_ENCODED" -d 'grant_type=client_credentials' | python -c 'import sys, json; print json.load(sys.stdin)["access_token"]')
+    ```
+    You can test the access token by calling the API to list the assemblies, for example:
+    ```
+    curl --insecure -H 'Accept: */*' -X GET -H "Authorization: Bearer $ACCESS_TOKEN" https://$PROXY_ADDRESS:443/api/topology/assemblies
+    ```
```

Comment:
Instead of running through the token generation in detail could we link to the Invoking REST API token similar to step 6 in the ansible example. Then use the $TNCO_URL and $TNCO_TOKEN in the curl command for adopt.
https://github.com/IBM/tnco-package-examples/blob/keys-certs/apache-ansible-demo/docs/adopt.md



5. Set up the environment to invoke the IBM Telco Network Cloud - Orchestration REST API.

Set the TNCO_URL environment variable to point to the API URL:
```
export TNCO_URL=<ishtar_route>
```
Set the TNCO_TOKEN environment variable to point to the access token
```
export TNCO_TOKEN=<access_token>
```
For more information about how to retrieve the ishtar_route and access_token parameters, see Invoking REST API (Add link here to the KC doc).